diff -Naru a/ogr/ogrsf_frmts/cad/libopencad/dwg/r2000.cpp b/ogr/ogrsf_frmts/cad/libopencad/dwg/r2000.cpp
--- a/ogr/ogrsf_frmts/cad/libopencad/dwg/r2000.cpp	2025-02-11 13:10:04 +0200
+++ b/ogr/ogrsf_frmts/cad/libopencad/dwg/r2000.cpp	2026-02-27 12:38:08 +0200
@@ -21,6 +21,7 @@
 #include <cstring>
 #include <iostream>
 #include <limits>
+#include <cstdint>
 #include <memory>
 #include <string>
 
diff -Naru a/swig/python/setup.py.in b/swig/python/setup.py.in
--- a/swig/python/setup.py.in	2025-02-11 13:10:04 +0200
+++ b/swig/python/setup.py.in	2026-02-27 12:45:42 +0200
@@ -12,6 +12,7 @@
 from glob import glob
 from pathlib import Path
 
+import setuptools
 from setuptools.command.build_ext import build_ext
 from setuptools import setup
 from setuptools import find_packages
@@ -157,7 +158,7 @@
 # since they may be directly used by users. So we do a monkey patching of
 # easy_install.install_wrapper_scripts to install a modified
 # easy_install.delete_blockers method that does NOT remove .py files
-if sys.platform == 'win32':
+if sys.platform == 'win32' and int(setuptools.__version__.split('.')[0]) < 80:
     from setuptools.command.easy_install import easy_install
 
     original_install_wrapper_scripts = easy_install.install_wrapper_scripts
